src/Features/Core/Portable/DesignerAttribute/DesignerAttributeDiscoveryService.cs

```diff
@@ -137,7 +137,7 @@ private static async ValueTask<bool> HasDesignerCategoryTypeAsync(Project projec
             // Freeze the entire solution at this point.  We don't want to run generators (as they are very unlikely
             // to contribute any changes that would affect which types we think are designable), and we want to be 
             // very fast to update the ui as a user types.
-            var frozenSolution = await solution.WithFrozenPartialCompilationsAsync(cancellationToken).ConfigureAwait(false);
+            var frozenSolution = solution.WithFrozenPartialCompilations(cancellationToken);
```

Comment:
became synchronous as this can be done entirely without async work.

Reply:
It looks like it was never async before, other than because of us calling the AsyncLazy GetValueAsync? I wonder if it's good/bad for us to be using AsyncLazy too for non-async cases when we want regular Lazy<T> but also have a bit friendlier of a cancellation story.

Reply:
yeah.  it's def interesting.  i don't personally mind it :)